CefSharp.WinForms/ChromiumWebBrowser.cs

```diff
@@ -273,25 +288,76 @@ private static void OnApplicationExit(object sender, EventArgs e)
         /// <summary>
         /// Initializes a new instance of the <see cref="ChromiumWebBrowser"/> class.
         /// </summary>
+        public ChromiumWebBrowser()
+        {
+
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChromiumWebBrowser"/> class.
+        /// </summary>
         /// <param name="address">The address.</param>
-        /// <exception cref="System.InvalidOperationException">Cef::Initialize() failed</exception>
         public ChromiumWebBrowser(string address)
         {
-            if (!Cef.IsInitialized && !Cef.Initialize())
+            Address = address;
+        }
+
+        private void Initialize()
+        {
+            if (!initialized)
             {
-                throw new InvalidOperationException("Cef::Initialize() failed");
-            }
+                if (!Cef.IsInitialized && !Cef.Initialize())
+                {
+                    throw new InvalidOperationException("Cef::Initialize() failed");
+                }
 
-            Cef.AddDisposable(this);
-            Address = address;
+                Cef.AddDisposable(this);
+
+                if (FocusHandler == null)
+                {
+                    FocusHandler = new DefaultFocusHandler(this);
+                }
+
+                if (ResourceHandlerFactory == null)
+                {
+                    ResourceHandlerFactory = new DefaultResourceHandlerFactory();
+                }
 
-            Dock = DockStyle.Fill;
+                if (BrowserSettings == null)
+                {
+                    BrowserSettings = new BrowserSettings();
+                }
+
+                managedCefBrowserAdapter = new ManagedCefBrowserAdapter(this, false);
+
+                initialized = true;
+            }
+        }
+
+        ~ChromiumWebBrowser()
+        {
+            if (!designMode)
+            {
+                Dispose(false);
+            }
+            else
+            {
+                base.Dispose(false);
+            }
+        }
 
-            FocusHandler = new DefaultFocusHandler(this);
-            ResourceHandlerFactory = new DefaultResourceHandlerFactory();
-            BrowserSettings = new BrowserSettings();
+        public new void Dispose()
```

Comment:
I only had to prevent the default Control.Dispose() from getting called.
I removed the `IDisposable` as it can be done without it.

> Was there a problem adding the design time checking logic to the existing protected override void Dispose(bool disposing)?

Yes, it crashed when I checked it in that function.
So I could either move the code out of that function or replace the default Dispose()